src/Symfony/Component/Form/Form.php

```diff
@@ -507,6 +508,10 @@ public function handleRequest($request = null)
      */
     public function submit($submittedData, $clearMissing = true)
     {
+        if ($submittedData instanceof Request) {
+            return $this->handleRequest($submittedData);
```

Comment:
It shouldn't throw the AlreadySubmittedException, because it follows a different execution path inside submit()

1 Form::submit(Request)
1.1 Form::handleRequest(Request)
1.1.1 HttpFoundationRequestHandler::handleRequest(Form, Request)
1.1.1.1 Form::submit(data)

only at: 1.1.1.1 Form::submitted = true

And if we used bind
preBindEvent dispatches (does not submit, so submitted is still false)
bind will call Form::submit(Request) which will follow the same execution path above.